In [295]:
import numpy as np
import math as m
import scipy.stats as stats
from scipy.stats import norm
from scipy.stats import t

Prob 1, Qs 1

Null Hypothesis: The Mean Engagement of Knowledgeable Students is (0.75).

Alternative Hypothesis: The Mean Engagement of Knowledgeable Students is not (0.75).

Test Used: Z-Test because (# of data points > 30), therefore Central Limit Theorem can be applied

Prob 1, Qs 2

In [296]:
myFile = open('engagement_1.txt')
data = myFile.readlines()
data = [float(x) for x in data]
size = len(data)
avg = np.mean(data)
sd = np.std(data, ddof = 1) / m.sqrt(size)
z = (avg - 0.75) / sd
p = 2 * norm.cdf(-abs(z))
print("Sample Size: {}".format(size))
print("Sample Mean: {}".format(avg))
print("Standard Error: {}".format(sd))
print("Standard Score: {}".format(z))
print("P-Value: {}".format(p))
myFile.close()

Sample Size: 937
Sample Mean: 0.7430304110448239
Standard Error: 0.004153027288269652
Standard Score: -1.6781948375012814
P-Value: 0.09330906925243751


The results are significant at a level of (0.1) because the P-Value is less than (0.1).
The results are NOT significant at a level of (0.05) because the P-Value is greater than
(0.05). The results are NOT significant at a level of (0.01) because the P-Value is
greater than (0.01). From this we can conclude that at (0.1) we reject the null
hypothesis and at (0.5) & (0.01) we fail to reject the null hypothesis.

Prob 1, Qs 3

In [297]:
size = len(data)
avg = np.mean(data)
sd = np.std(data, ddof = 1) / m.sqrt(size)
p = 2 * norm.cdf(-abs(z))
while p >= 0.05:
    size+=1
    sd = np.std(data, ddof = 1) / m.sqrt(size)
    z = (avg - 0.75) / sd
    p = 2 * norm.cdf(-abs(z))
print("Largest Standard Error: {}".format(sd))
print("Corresponding Minimum Sample Size: {}".format(size))

Largest Standard Error: 0.0035546701555212953
Corresponding Minimum Sample Size: 1279


The largest standard error for which the test will be significant at a level of 0.05 is (0.0499).
The corresponding minimum sample size is (1279).

Prob 1, Qs 4

Null Hypothesis: The Mean Engagement of Knowledgeable Students is equivalent to The Mean Engagement of not knowledgeable Students.

Alternative Hypothesis: The Mean Engagement of Knowledgeable Students is not equivalent to The Mean Engagement of  not knowledgeable Students.
Test Used: Z-Test because (# of data points > 30), therefore Central Limit Theorem can be applied.

Prob 1, Qs 5

In [298]:
myFile1 = open('engagement_1.txt')
data1 = myFile1.readlines()
myFile2 = open('engagement_0.txt')
data2 = myFile2.readlines()
data1, data2 = [float(x) for x in data1], [float(y) for y in data2]
size1, size2 = len(data1), len(data2)
avg1, avg2 = np.mean(data1), np.mean(data2)
sd1 = (np.std(data1, ddof = 1)**2) / size1
sd2 = (np.std(data2, ddof = 1)**2) / size2
sd = m.sqrt(sd1 + sd2)
z = (avg2 - avg1) / sd
p = 2 * norm.cdf(-abs(z))
print("For Engagement_1, Sample Size: {}, Sample Mean: {}".format(size1,avg1))
print("For Engagement_0, Sample Size: {}, Sample Mean: {}".format(size2, avg2))
print("Standard Error: {}".format(sd))
print("Z-Score: {}".format(z))
print("P-Value: {}".format(p))
myFile1.close()
myFile2.close()

For Engagement_1, Sample Size: 937, Sample Mean: 0.7430304110448239
For Engagement_0, Sample Size: 1977, Sample Mean: 0.6399545077035914
Standard Error: 0.007065420910043284
Z-Score: -14.588784540028351
P-Value: 3.3104307168195455e-48


If the P-Value is less than the level, we Reject the Null Hypothesis
Therefore, for levels (0.1), (0.05) and (0.01), we Reject the Null Hypothesis.
Because of this, we can draw a similar conclusion for all these levels that
the mean engagement of knowledgeable students is not the same as those not
knowledgeable.

Prob 2, Qs 1

A T-Test will be used.

In [299]:
data = np.array([3, -3, 3, 12, 15, -16, 17, 19, 23, -24, 32])
avg = np.mean(data)
size = len(data)
t_c = t.ppf(.95, size - 1)
sd = np.std(data, ddof = 1) / m.sqrt(size)
lower_b, upper_b = avg - (sd * t_c), avg + (sd * t_c)
print("T-Value: {}".format(t_c))
print("90% Confidence Interval: ({}, {})".format(lower_b, upper_b))

T-Value: 1.8124611228107335
90% Confidence Interval: (-1.8369195722533416, 16.56419229952607)


Prob 2, Qs 2

A T-Test will be used.

In [300]:
t_c = t.ppf(.975, size - 1)
lower_b, upper_b = avg - (sd * t_c), avg + (sd * t_c)
print("Sample Mean: {}".format(avg))
print("Standard Error: {}".format(sd))
print("T-Value: {}".format(t_c))
print("95% Confidence Interval: ({}, {})".format(lower_b, upper_b))

Sample Mean: 7.363636363636363
Standard Error: 5.0762776757504415
T-Value: 2.2281388519649385
95% Confidence Interval: (-3.9470151490654715, 18.674287876338198)


Compared to the 90% Confidence Interval, the 95% Confidence Interval is bigger.

Prob 2, Qs 3

A Z-Test will be used.

In [301]:
data = np.array([3, -3, 3, 12, 15, -16, 17, 19, 23, -24, 32])
avg = np.mean(data)
size = len(data)
z_c = norm.ppf(.975)
sd = 16.836/ m.sqrt(size)
lower_b, upper_b = avg - (sd * z_c), avg + (sd * z_c)
print("Standard Error: {}".format(sd))
print("Z-Value: {}".format(z_c))
print("95% Confidence Interval: ({}, {})".format(lower_b, upper_b))

Standard Error: 5.076244997311228
Z-Value: 1.959963984540054
95% Confidence Interval: (-2.585621007795268, 17.312893735067995)


Some Notable Differences:
- Standard error is smaller in Q3.
- Z-Value is less than the T-Value from Q1.
- 95% Confidence Interval is smaller in Q3.

Prob 2, Q4

In [302]:
avg = np.mean(data)
size = len(data)
c = 0.95
sd = np.std(data, ddof = 1) / m.sqrt(size)
t_c = stats.t.ppf(1-(1-c)/2, size- 1)
lower_b = avg - (sd * t_c)
while lower_b <= 0:
    t_c = stats.t.ppf(1-(1-c)/2, size - 1)
    lower_b = avg - (sd * t_c)
    c -= 0.00001
c_level = c * 100
print("With a confidence of {}%, we can say that the team is expected to win on average".format(c_level))


With a confidence of 82.24600000005805%, we can say that the team is expected to win on average
